packages/core/src/render3/reactivity/effect.ts

```diff
@@ -274,9 +275,11 @@ export const BASE_EFFECT_NODE: Omit<EffectNode, 'fn' | 'destroy' | 'injector'> =
     },
 
     maybeCleanup(this: EffectNode): void {
-      while (this.cleanupFns?.length) {
-        this.cleanupFns.pop()!();
-      }
+      untracked(() => {
+        while (this.cleanupFns?.length) {
+          this.cleanupFns.pop()!();
+        }
+      });
```

Comment:
I don't have a feeling for how many effects have cleanup logic, but might be meaningful to avoid untracking if there is no work to begin with:

```suggestion
      const cleanupFns = this.cleanupFns;
      if (cleanupFns) {
        untracked(() => {
          while (cleanupFns.length) {
            cleanupFns.pop()!();
          }
        });
      }
```